perses/analysis/visualization.py

```diff
@@ -0,0 +1,416 @@
+"""
+Visualization tools for perses (protein mutations or small molecule transformations).
+Requires PDB and trajectory (.dcd or .xtc) files for old and new atoms.
+
+Pre-requisites:
+conda install -c schrodinger pymol (+ PyMOL license)
+conda install -c conda-forge moviepy
+
+"""
+
+__author__ = 'Ivy Zhang'
+
+################################################################################
+# IMPORTS
+################################################################################
+
+import sys, os
+
+try:
+    import pymol
+    from pymol import cmd
+    _FOUND_PYMOL = True
+except:
+    _FOUND_PYMOL = False
+
+try:
+    import moviepy.editor as mpy
+    _FOUND_MPY = True
+except:
+    _FOUND_MPY = False
+
+def _check_pymol():
+    """Check whether pymol was imported, if not raise exception. """
+    if _FOUND_PYMOL:
+        return
+    else:
+        raise ImportError("PyMOL is required for this module. Please `conda install -c schrodinger pymol`.")
+
+def _check_mpy():
+    """Check whether moviepy was imported, if not raise exception. """
+    if _FOUND_MPY:
+        return
+    else:
+        raise ImportError(
+            "moviepy is required for this module. Please `conda install -c conda-forge moviepy` or `pip install moviepy`.")
```

Comment:
I think it could make this clearer and less prone to errors if we deal the except directly with a `ModuleNotFoundError`.
```suggestion
try:
    import pymol
    from pymol import cmd
except ModuleNotFoundError as error:
   raise error("PyMOL is required for this module. Please `conda install -c schrodinger pymol`.")

try:
    import moviepy.editor as mpy
except ModuleNotFoundError as error:
    raise error("moviepy is required for this module. Please `conda install -c conda-forge moviepy` or `pip install moviepy`.")
```

Reply:
The reason why I didn't write it with a `ModuleNotFoundError` directly in the `except` is because we don't want to force `pymol` and `moviepy` to be dependencies for perses. Therefore, I only want there to be a `ModuleNotFoundError` or `ImportError` when someone actually tries to use this module (aka when they instantiate a `Visualization` object). The current implementation for throwing the `ImportError` was suggested by Levi Naden (software scientist at molssi), but if you can think of a cleaner way to do it, please let me know!

Reply:
Oh okay, I understand. There is probably a cleaner way, but I guess this works for now, that's why it was a non-blocking suggestion. No worries.

Reply:
What if instead we just imported `pymol` and `moviepy` instead the `Visualization` class? That would prevent it from being a dependency and simplify this logic.

Reply:
I don't think that will work, because then pymol and moviepy will only be available in the init, and not for the rest of the functions.